### PR TITLE
typing any for Scatterplot and Lineplot Vizs

### PR DESCRIPTION
**Pull request by moontrip**
This will address https://github.com/VEuPathDB/web-eda/issues/1488.

It would be great if you can review and make any suggestions, if necessary, @dmfalke 👍

### CODE REVIEW

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx [hunk 1]
```diff
@@ -198,6 +198,29 @@ export interface ScatterPlotDataWithCoverage extends CoverageStatistics {
 // define ScatterPlotDataResponse
 type ScatterPlotDataResponse = ScatterplotResponse;
 
+// define dataSetProcess type used inside the processInputData()
+// Note that dataSetProcess is different from the one used in the outside of the processInputData()
+// using pre-existing ScatterPlotDataSeries to override
+type Override<T1, T2> = Omit<T1, keyof T2> & T2;
```
Comment (dmfalke) [inline packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:204]: This is a nice utility type. Can you add it to `packages/libs/eda/src/lib/core/types/utility.ts`?

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx [hunk 2]
```diff
@@ -198,6 +198,29 @@ export interface ScatterPlotDataWithCoverage extends CoverageStatistics {
...
+type dataSetProcessType = Override<
```
Comment (dmfalke) [inline packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:205]: Please start types with an uppercase letter
```suggestion
type DataSetProcessType = Override<
```

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx [hunk 3]
```diff
@@ -970,7 +970,7 @@ function LineplotViz(props: VisualizationProps<Options>) {
     }
     return Object.entries(dataElementConstraints[0])
       .filter((variable) => variable[1].isRequired)
-      .every((reqdVar) => !!(vizConfig as any)[reqdVar[0]]);
+      .every((reqdVar) => !!vizConfig[reqdVar[0] as keyof typeof vizConfig]);
```
Comment (dmfalke) [inline packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx:973]: I think you can use `LineplotConfig` here
```suggestion
      .every((reqdVar) => !!vizConfig[reqdVar[0] as keyof LineplotConfig]);
```

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx [hunk 4]
```diff
@@ -1990,7 +2013,7 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
     if (!dataElementConstraints) return false;
     return Object.entries(dataElementConstraints[0])
       .filter((variable) => variable[1].isRequired)
-      .every((reqdVar) => !!(vizConfig as any)[reqdVar[0]]);
+      .every((reqdVar) => !!vizConfig[reqdVar[0] as keyof typeof vizConfig]);
```
Comment (dmfalke) [inline packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:2016]: I think you can use `ScatterplotConfig` here
```suggestion
      .every((reqdVar) => !!vizConfig[reqdVar[0] as keyof ScatterplotConfig]);
```

**Review by dmfalke — CHANGES_REQUESTED** [review]
This is really nice. I requested a few changes. Let me know what you think.

**Comment by moontrip** [comment]
Thank you for your detailed comments and suggestions, @dmfalke 👍 I will probably take care of them tomorrow (technically today as it passed 12AM 😅)

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx (hunk 1 above)
Comment (moontrip) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx]: @dmfalke Yep, I have complied 😃

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx (hunk 2 above)
Comment (moontrip) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx]: Oops! I thought it was capital :( Changed it

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx (hunk 3 above)
Comment (moontrip) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx]: 👍 complied

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx (hunk 4 above)
Comment (moontrip) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx]: 👍 complied

**Comment by moontrip** [comment]
@dmfalke Thanks again for valuable comments. All feedbacks are addressed~

**Review by dmfalke — APPROVED** [review]
Looks great!

**Pull request merged**